enhancements/api-review/custom-tags-aws.md

```diff
@@ -79,65 +101,373 @@ The userTags field is intended to be set at install time and is considered immut
+
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource.
+
+#### Delete tags scenarios
+Tags are deleted when the user sets the user-defined tag value to an empty string in `.spec.platformSpec.aws.resourceTags`.
+Also refer to `Precedence` scenario to understand cases where deletion of user-defined tags is not allowed .
+
+For example,
+
+Existing tag for AWS resource = `key_infra1 = value_old`
+
+New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 =`
+
+Action = Existing tag for AWS resource is deleted.
+
+Final tag set to AWS resource = deleted
+
+Event action = An event is generated to notify user about the action status (success/failure) to delete tags for the AWS resource.
+
+#### Precedence scenarios
+1) User-defined tags on kube resources MUST continue to take precedence during create and update. User-defined tags found on kube resources must not be deleted by methods described in `Delete tags scenarios`.
+   A warning is generated to inform the user about action not being applied on the list of user tags.
+
+   For example,
+
+   Existing tag in kube resource object like machine CR = `key_infra1 = custom_value`
+
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value1`
+
+   Action = The tag for AWS resource is maintained to `key_infra1 = custom_value`.
+
+   Final tag set to AWS resource = `key_infra1 = custom_value`
+
+   Event action = An event is generated to notify user about the action status (failure) to update tags for the AWS resource. A warning also must be generated about list of user-defined tags on which action is not applicable.
+
+2) When a new user-defined tag is added on kube resources, it MUST override the tag in `.spec.platformSpec.aws`. User-defined tags found on kube resources must not be deleted by methods described in `Delete tags scenarios`.
+   A warning is generated to inform the user about action not being applied on the list of user tags.
+
+   For example,
+
+   Existing tag in Infrastructure CR = `key_infra1 = custom_value`
+
+   New tag added in kube resource object like machine CR = `key_infra1 = value1`
+
+   Action = The tag for AWS resource is maintained to `key_infra1 = value1`.
+
+   Final tag set to AWS resource = `key_infra1 = custom_value`
+
+   Event action = An event is generated to notify user about the action status (success) to update tags for the AWS resource. A warning also must be generated about list of user-defined tags on which action did a override.
+
+3) `.spec.platformSpec.aws` take precedence over `.status.platformStatus.aws`. User-defined tags must be merged from `.status.platformStatus.aws` and set to AWS resource.
+
+4) When `.status.platformStatus.aws` has a same user-defined tag as in `.spec.platformSpec.aws`,the entry must be removed from `.status.platformStatus.aws` before deleting user-defined tag from AWS resource.
```

Comment:
The language of this sentence is ambiguous. What does it mean that "the entry must be removed"? Does it mean that the operation of changing the infrastructure resource will fail? Does it mean that there is undefined behavior? Does it mean that some other action will not happen until the entry is removed?

Reply:
Rephrased to add more on user usage  PoV.